specs/dispute-game-interface.md

```diff
@@ -24,53 +25,188 @@ dispute games to exist. If multiple dispute games run in production, it gives
     function version() external pure returns (string memory _version);
 
+    /// @notice Returns the timestamp that the DisputeGame contract was created at.
+    function createdAt() external pure returns (Timestamp _createdAt);
+
+    /// @notice Returns the current status of the game.
+    function status() external view returns (GameStatus _status);
+
+    /// @notice `clones-with-immutable-args` argument #1
     /// @return _gameType The type of proof system being used.
     /// @dev The reference impl should be entirely different depending on the type (fault, validity)
-    /// i.e. The game type should indicate the security model.
+    ///      i.e. The game type should indicate the security model.
     function gameType() external view returns (GameType _gameType);
-}
-```
 
-The generic dispute game interface is extended to work specifically with a fault
-proof or a validity proof since their modes of operation are quite different.
+    /// @notice `clones-with-immutable-args` argument #2
+    /// @return _rootClaim The root claim of the DisputeGame.
+    function rootClaim() external view returns (Claim _rootClaim);
 
-### Disputable Interface
+    /// @notice `clones-with-immutable-args` argument #3
+    /// @return _extraData Any extra data supplied to the dispute game contract by the creator.
+    function extraData() external view returns (bytes memory _extraData);
 
-Any contract that can be disputed by a dispute game implementation should
-implement the disputable interface. This allows for any dispute game to
-successfully interact with a variety of contracts.
+    /// @notice Returns the address of the `BondManager` used 
+    function bondManager() public view returns (IBondManager _bondManager);
 
-```solidity
-/// @title IDisputable
-/// @notice The generic interface for a disputable contract
-interface IDisputable {
-    /// @notice The implementation of the DisputeGame calls this after a
-    /// successful dispute. This should be guarded such that it is only
-    /// callable by a DisputeGame.
-    function challenge(bytes memory _data) external;
+    /// @notice If all necessary information has been gathered, this function should mark the game
+    ///         status as either `CHALLENGER_WINS` or `DEFENDER_WINS` and return the status of
+    ///         the resolved game. It is at this stage that the bonds should be awarded to the
+    ///         necessary parties.
+    /// @dev May only be called if the `status` is `IN_PROGRESS`.
+    function resolve() public returns (GameStatus _status);
+}
+
+////////////////////////////////////////////////////////////////
+//              OUTPUT ATTESTATION DISPUTE GAME               //
+////////////////////////////////////////////////////////////////
+
+/// @title IDisputeGame_OutputAttestation
+/// @notice The interface for an attestation-based DisputeGame meant to contest output
+///         proposals in Optimism's `L2OutputOracle` contract.
```

Comment:
The `L2OutputOracle` will likely be suffixed with `V2`. So, can just do this

```suggestion
///         proposals in Optimism's `L2OutputOracleV2` contract.
```

Reply:
I believe that once the `L2OutputOracleV2` changes are done, we'll just merge it into the `L2OutputOracle` contract and update Semver.

Reply:
Yea that's cool by me :)